src/_blog/libp2p-webrtc-browser-to-server.md

```diff
@@ -0,0 +1,248 @@
+
+```proto
+message Message {
+  enum Flag {
+    FIN = 0;
+    STOP_SENDING = 1;
+    RESET_STREAM = 2;
+  }
+  optional Flag flag=1;
+  optional bytes message = 2;
+}
+```
+
+#### Multiaddress
+
+The [multiaddress](https://docs.libp2p.io/concepts/fundamentals/addressing/) of a WebRTC address begins like a standard UDP address, but adds 3 additional components: `webrtc`, `hash`, and `p2p`.
+
+```shell
+/ip4/1.2.3.4/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>
+```
+
+* `webrtc`: the name of this transport
+* `hash`: the [multihash](https://github.com/multiformats/multihash) of the certificate used in the DTLS handshake
+* `p2p`: the peer-id of the libp2p node (optional)
+
+### Benefits
+
+#### Self-signed Certificate
+
+WebRTC enables browsers to connect to public libp2p nodes without the nodes requiring a TLS certificate in the browser's [certificate chain](https://en.wikipedia.org/wiki/X.509#Certificate_chains_and_cross-certification). Because the server can use a self-signed TLS certificate, WebRTC removes the need to include additional services like DNS and Let's Encrypt. 
+
+#### Broad support
+
+Chrome has supported WebRTC since 2012.  Other browsers soon followed, achieving support [on all evergreen browsers](https://caniuse.com/?search=webrtc).  WebRTC is literally everywhere.
+
+### Limitations
+
+#### Setup and configuration
+
+Because WebRTC represents a collection of technologies, it requires extensive setup and configuration compared to other transports.  
+
+#### Extensive Roundtrips
+
+Another limitation is the 6 roundtrips required before data is exchanged.  This makes WebTransport a more compelling transport for the browser to server use case where the browser supports WebTransport.
+
+### Usage
+
+The complexity of WebRTC is abstracted in the implementations, making it seamless to swap your existing transport with WebRTC.
+
+Let's look at the JavaScript implementation as an example:
+
+```javascript
+import { webRTC } from 'js-libp2p-webrtc'
+
+const node = await createLibp2p({
+  transports: [webRTC()],
+  connectionEncryption: [() => new Noise()],
+});
+```
+
+The only difference from other transports is initializing with `webRTC()`.  That's all you need to do to implement WebRTC in the browser.  Easy, right?
+
+
+## Alternative transports
+
+WebRTC is one of many ways to connect browsers to a libp2p node. [Choosing the transport](https://connectivity.libp2p.io/) that fits your use case is one of the many unique strengths of libp2p.
+
+### WebSocket
+
+The [WebSocket RFC](https://datatracker.ietf.org/doc/html/rfc6455) dates back to 2011 and specifies the opening of a two-way socket from a browser to a server over TCP. WebSocket is implemented in the [Rust](https://github.com/libp2p/rust-libp2p/tree/master/transports/websocket), [Go](https://github.com/libp2p/go-libp2p/tree/master/p2p/transport/websocket), and [JavaScript](https://github.com/libp2p/js-libp2p-websockets) libp2p implementations. 
```

Comment:
Isn't a benefit that it is supported by Rust, next to also Go and Js, while WebTransport doesn't support Rust (yet)?

Reply:
I agree @GlenDC, though there was a request to thin out the WS section.  Maybe this could be a limitation of WebTransport?